mitiq/lre/lre.py

```diff
@@ -46,10 +47,14 @@ def execute_with_lre(
 
     Args:
         input_circuit: Circuit to be scaled.
-        executor: Executes a circuit and returns a `float`
+        executor: Executes a circuit and returns a `float`.
         degree: Degree of the multivariate polynomial.
         fold_multiplier: Scaling gap value required for unitary folding which
             is used to generate the scale factor vectors.
+        observable: Observable to compute the expectation value of. If
+            ``None``, the ``executor`` must return an expectation value.
+            Otherwise, the ``float`` returned by ``executor`` is used
+            to compute the expectation of the observable.
```

Comment:
```suggestion
        observable: Observable to compute the expectation value of. If
            ``None``, the ``executor`` must return an expectation value.
            Otherwise, the ``MeasurementResult`` returned by ``executor`` is used
            to compute the expectation of the observable.
```

Should this be measurement result or density matrix?

Reply:
Density matrix would make more sense as discussed [here](https://github.com/unitaryfund/mitiq/issues/2513#issuecomment-2691225754).

Reply:
When an observable is passed, the executor should also be able to return bitstrings **or** a density matrix.

Reply:
Then maybe I should write "the ``DensityMatrix`` or ``Bitstrings`` returned". Because writing ``MeasurementResult`` does not indicate the type of object being returned.

Reply:
Agreed!